### PR TITLE
Drop support for ruby 2.3

### PR DESCRIPTION
**Pull request by apolcyn**
Towards https://github.com/grpc/grpc/issues/25194

Plan to followup to drop support for ruby 2.4 in the next few months

### CODE REVIEW

File: grpc.gemspec [hunk 1]
```diff
@@ -13,7 +13,7 @@ Gem::Specification.new do |s|
   s.description   = 'Send RPCs from Ruby using GRPC'
   s.license       = 'Apache-2.0'
 
-  s.required_ruby_version = '>= 2.3.0'
+  s.required_ruby_version = '>= 2.4.0'
```
Comment (stanley-cheung) [inline grpc.gemspec:16]: does this needs to be fixed in the `templates/` directory too?

**Review by stanley-cheung — APPROVED** [review]

**Review by jtattermusch — APPROVED** [review]
LGTM.

File: grpc.gemspec (hunk 1 above)
Comment (apolcyn) [reply grpc.gemspec:16]: correct, this has been fixed

**Comment by apolcyn** [comment]
cloud to cloud test failure: https://github.com/grpc/grpc/issues/25090

**Comment by apolcyn** [comment]
basic tests c windows passed in https://source.cloud.google.com/results/invocations/038ec017-a811-4962-af4e-8b335694afec/targets, but the result is still shown as "pending" here

**Pull request merged**